hal/src/nRF52840/core_hal.c

```diff
@@ -541,6 +542,9 @@ int HAL_Core_Enter_Stop_Mode_Ext(const uint16_t* pins, size_t pins_count, const
         }
     }
 
+    // Disable thread scheduling
+    os_thread_scheduling(false, NULL);
```

Comment:
This shouldn't be done. You are disabling scheduling and later on attempt to acquire the exflash lock which is implemented using a recursive mutex. If by this point it's being held by some other thread, it will never be released and everything will deadlock. `sd_nvic_critical_region_enter` will also disable thread scheduling, disabling the scheduler is unnecessary.